tests/test_preprocessing/test_main.py

```diff
@@ -30,3 +32,28 @@ def test_main(tmp_path, files, mocker):
         runpy.run_module("torch_dicom.preprocessing", run_name="__main__", alter_sys=True)
     except SystemExit as e:
         raise e.__context__ if e.__context__ is not None else e
+
+
+@pytest.mark.parametrize(
+    "pattern",
+    [
+        "ERROR: Invalid value for VR: ",
+        "The Bits Stored Value does not match",
+    ],
+)
+def test_main_with_warnings(mocker, pattern, capsys):
+    def side_effect_function(_):
+        warnings.warn(pattern)
+        print("Warning issued")
+
+    mocker.patch("torch_dicom.preprocessing.__main__.main", side_effect=side_effect_function)
+    mocker.patch("torch_dicom.preprocessing.__main__.parse_args")
+
+    with warnings.catch_warnings(record=True) as w:
+        warnings.simplefilter("always")
+        entrypoint()
+
+    captured = capsys.readouterr()
+    assert not any(pattern in str(warning.message) for warning in w)
```

Comment:
Should we add a "control" pattern that we assert is present and not filtered out? Such as:
```python
assert any(control_pattern in str(warning.message) for warning in w)
```